[212] - Add request validation to EmailController

RockTF opened the pull request:
[212] - [Add request validation to EmailController](https://github.com/OlhenShu/ProjectStage_Team1_May/issues/212)

- [x] Obtained approval from the owner
- [x] Made changes to the code
- [x] Requested a review

RockTF committed:
issues/212 - Add request validation to EmailController

Added data validation to the 'changePlaceStatus' endpoint in EmailController. If the request body does not pass validation, an error response is returned. Additionally, annotations for email validation were added in SendChangePlaceStatusEmailMessage model.

RockTF committed:
issues/212 - Replace @Validated with @Valid in EmailController

The @Validated annotation from SpringFramework was replaced by @Valid annotation from Jakarta in the EmailController class. This change was made specifically in the method changePlaceStatus to perform input validation.

BerserkrMM commented on core/src/main/java/greencity/controller/EmailController.java:
```diff
@@ -56,10 +58,24 @@ public ResponseEntity<Object> sendReport(@RequestBody SendReportEmailMessage mes
+            BindingResult bindingResult
+    ) {
+        if (bindingResult.hasErrors()) {
+            return ResponseEntity.badRequest().body(bindingResult.getAllErrors());
```
Perhaps better to throw BadRequest and give responsibility to ExceptionHandler to build ResponseEntity?

RockTF replied:
Right. If errors are handled by an error handler, it would be better to transfer this responsibility to it.

IliaWithHat reviewed (commented):
Don't use BindingResult. Just remove it. And import in Intellij Idea checkstyle.xml because there should be 4 spaces before @ApiResponse, not 8.

nataliiapovzhyk reviewed (approved):
tests passed.no errors in Sonarlint in updated files

RockTF committed:
Merge branch 'refs/heads/dev' into issues/212

# Conflicts:
#	core/src/main/java/greencity/controller/EmailController.java

RockTF committed:
Refactor code to improve readability

Cleaned up indentation and whitespace for various test and main files, ensuring uniformity across the codebase. This commit is purely cosmetic, does not involve any functional changes, and is intended to reflect a more maintainable standard for code structure.

RockTF committed:
Merge remote-tracking branch 'origin/dev' into issues/212

# Conflicts:
#	core/src/main/java/greencity/config/SecurityConfig.java
#	core/src/main/java/greencity/controller/UserController.java
#	service/src/main/java/greencity/service/EmailServiceImpl.java

RockTF committed:
Merge remote-tracking branch 'origin/dev' into issues/212

# Conflicts:
#	core/src/test/java/greencity/controller/UserSecuredControllerTest.java

RockTF committed:
Update UserSecuredControllerTest for various endpoints

This commit includes the update of tests in UserSecuredControllerTest. The updated tests cover response status for different endpoints like deleting profile picture, checking online status and get endpoint as authenticated ADMIN. It verifies the correct HTTP status based on the user's role and action.

RockTF committed:
Update UserSecuredControllerTest for various endpoints

This commit includes the update of tests in UserSecuredControllerTest. The updated tests cover response status for different endpoints like deleting profile picture, checking online status and get endpoint as authenticated ADMIN. It verifies the correct HTTP status based on the user's role and action.

RockTF committed:
Refactor NewsSubscriberRepo interface

Removed unnecessary spaces in the NewsSubscriberRepo interface. This makes the code look cleaner and ensures consistency across the codebase.

RockTF committed:
Refactor code for better readability and format compliance

The changes are mainly focused around code indentation, consistent bracket placement, and removing extra white spaces. Changes are performed in multiple files like Controllers, Services, Entity and Test classes to maintain clean and professional code standard.

RockTF committed:
Refactor imports and whitespace in multiple files

This commit reorders and refactors import statements in several files for better readability and adherence to project structure. It also addresses white space issues by removing extra lines and ensuring newline at the end of each file.

RockTF committed:
Reformat import statements and spaces in DTO classes

Reorganized the import statements and eliminated excessive spacing in both NewsSubscriberRequestDto and SendNewsDto. The changes allow to improve readability, while no functional modifications were made in the code.

RockTF committed:
Remove unnecessary whitespace in NewsSubscriberRequestDto

An unneeded line break was eliminated in the NewsSubscriberRequestDto file. This contributes to cleaner, more readable code and adheres to optimal coding practices.
